src/nwb_datajoint/decoding/clusterless.py

```diff
@@ -156,3 +164,304 @@ def _convert_to_dataframe(nwb_data):
                             index=pd.Index(nwb_data['marks'].timestamps,
                                            name='time'),
                             columns=columns)
+
+    @staticmethod
+    def _get_peak_amplitude(waveform):
+        spike_peak_ind = waveform.shape[1] // 2
+        return waveform[:, spike_peak_ind]
+
+    @staticmethod
+    def _threshold(timestamps, marks, mark_param_dict):
+        # filter the marks by the amplitude threshold
+        if mark_param_dict['sign'] == -1:
+            threshold = (mark_param_dict['sign'] *
+                         mark_param_dict['threshold'])
+            include = np.min(marks, axis=1) <= threshold
+        elif mark_param_dict['sign'] == 1:
+            threshold = (mark_param_dict['sign'] *
+                         mark_param_dict['threshold'])
+            include = np.max(marks, axis=1) >= threshold
+        else:
+            threshold = mark_param_dict['sign']
+            include = np.max(np.abs(marks), axis=1) >= threshold
+        return timestamps[include], marks[include]
+
+
+@schema
+class UnitMarksIndicatorSelection(dj.Lookup):
+    definition = """
+    -> UnitMarks
+    -> IntervalList
+    sampling_rate=500 : float
+    ---
+    """
+
+
+@schema
+class UnitMarksIndicator(dj.Computed):
```

Comment:
what is a unit marks indicator?

Reply:
is it just marks with interval?

Reply:
It is the marks binned into 2 ms time bins. I can add a description.